griptape/drivers/web_search/tavily_web_search_driver.py

```diff
@@ -0,0 +1,37 @@
+from __future__ import annotations
+
+import json
+from typing import TYPE_CHECKING
+
+from attrs import Factory, define, field
+
+from griptape.artifacts import ListArtifact, TextArtifact
+from griptape.drivers import BaseWebSearchDriver
+from griptape.utils import import_optional_dependency
+
+if TYPE_CHECKING:
+    from tavily import TavilyClient
+
+
+@define
+class TavilyWebSearchDriver(BaseWebSearchDriver):
+    api_key: str = field(kw_only=True)
+    client: TavilyClient = field(
+        default=Factory(lambda self: import_optional_dependency("tavily").TavilyClient(self.api_key), takes_self=True),
+        kw_only=True,
+    )
+
+    def search(self, query: str, **kwargs) -> ListArtifact:
+        try:
+            response = self.client.search(query, max_results=self.results_count, **kwargs)
+            results = response["results"]
+            return ListArtifact(
+                [
+                    TextArtifact(
+                        json.dumps({"title": result["title"], "url": result["url"], "content": result["content"]})
+                    )
+                    for result in results
+                ]
+            )
+        except Exception as e:
+            raise Exception(f"Error searching '{query}' with Tavily: {e}") from e
```

Comment:
I know the other Drivers do it, but catching `Exception` and then immediately re-raising it does not provide much value. @william-price01 are there any more specific exceptions we can catch?

Reply:
Maybe a missing API-Key? Not really sure. Most of the errors are already handled by the Tavily SDK, they have built in Exception types. Maybe I utilize those?

Reply:
not sure that catching specific exceptions just to print and reraise is much better either. should we just remove the `try/except` and let the tool handle it?